src/main/java/io/codearte/jfairy/producer/person/locale/zh/ZhNationalIdentityCardNumberProvider.java

```diff
@@ -0,0 +1,113 @@
+package io.codearte.jfairy.producer.person.locale.zh;
+
+import com.google.inject.Inject;
+import io.codearte.jfairy.producer.BaseProducer;
+import io.codearte.jfairy.producer.DateProducer;
+import io.codearte.jfairy.producer.TimeProvider;
+import io.codearte.jfairy.producer.person.NationalIdentityCardNumberProvider;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import java.lang.management.ManagementFactory;
+import java.util.Random;
+
+
+/**
+ * Chinese National Identity Card Number
+ *
+ * @author Lhfcws
+ * @since 27.02.17
+ */
+public class ZhNationalIdentityCardNumberProvider implements NationalIdentityCardNumberProvider {
+	/**
+	 * Codes of China provinces
+	 */
+	private static final String[] PROV_LIST = {
```

Comment:
Please change your local variables such as s, r, etc., to have more miningful names.